### PR TITLE
Les-39-Password

### PR DESCRIPTION
**Pull request by Kurilo-Pavel**

### CODE REVIEW

File: src/FormInput.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        display: 'inline-block',
+        position: 'relative',
+        marginBottom: '-0.5em',
+        width: '1.5em',
+        height: '1.5em',
+        borderRadius: '50%',
+        marginLeft: '-2em'
+      }} onClick={e => showPassword(e)}/>}
+      {(fieldError && fieldBlur) && <div style={{color: 'red'}}>{label} entered incorrectly</div>}
+    </div>
```
Comment (Evgeniy121212) [inline src/FormInput.js:68]: don't use inline styles - use `className` and css-classes for that

File: src/FormInput.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  return (
+    <div style={{margin: '20px 0'}}>
+      <label style={{display: 'block', fontWeight: 'bold', margin: '5px 0'}}>{label}</label>
+      <input onChange={e => fieldHandler(e)}
+             onBlur={e => blurHandler(e)}
+             name={label}
+             value={field}
+             type={type}
```
Comment (Evgeniy121212) [inline src/FormInput.js:49]: `type={ isOpen ? "text" : "password" }`

where `isOpen` - value from `useState`-hook, and settled at the `showPassword`-function

File: src/Registration.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import FormInput from "./FormInput";
```
Comment (Evgeniy121212) [inline src/Registration.js:1]: Please, put that file to the folder `/src/components/PasswordWithConfirm` and make it .tsx

File: src/Registration.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        <FormInput label={'Password'}
+                   validation={passwordRegex}
+                   type={'password'}
+                   required={'required'}
+                   name={'hidden'}/>
+        <FormInput label={'Confirm password'}
+                   type={'password'}
+                   required={'required'}
+                   name={'hidden'}/>
+      </form>
```
Comment (Evgeniy121212) [inline src/Registration.js:18]: 1. form will be included into the another ticket
2. password "with open-close eyes" not used when used "password" and "confirm password" fields

File: src/style/style.scss [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.modal{
+  width: 25%;
+  margin-left: auto;
+  margin-right: auto;
+  border: 4px solid #0B456F;
+  padding: 30px;
+}
```
Comment (Evgeniy121212) [inline src/style/style.scss:7]: please, remove that file:
1. it should be at the project, but into the another place
2. modal is not included into your ticket

File: src/components/Password.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <div className={'password'}>
+      <label className={'passwordLabel'}>Password</label>
+      <input type={visiblePassword ? 'text' : 'password'}
+             className={'passwordInput ' + `${(isError && isBlur) ? 'errorInput' : ''}`}
+             onChange={fieldHandler}
+             onBlur={blurHandle}
+             required/>
+      <img className={'image'} alt='eye' src={visiblePassword ? open_eye : close_eye} onClick={showPassword}/>
+      {(isError && isBlur) && <span className={'error'}>Password must contain from 6 to 256 characters</span>}
```
Comment (NatashaMikhaltsova) [inline src/components/Password.tsx]: store error messages in state

File: src/components/Password.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  const showPassword = (e: React.MouseEvent<HTMLImageElement>): void => {
+    if (visiblePassword !== false) {
+      setVisiblePassword(false);
+    } else {
+      setVisiblePassword(true);
+    }
+  }
+
+  return (
+    <div className={'password'}>
```
Comment (NatashaMikhaltsova) [inline src/components/Password.tsx]: apply classnames like className='password' without {}

File: src/components/Password.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      <label className={'passwordLabel'}>Password</label>
+      <input type={visiblePassword ? 'text' : 'password'}
+             className={'passwordInput ' + `${(isError && isBlur) ? 'errorInput' : ''}`}
+             onChange={fieldHandler}
+             onBlur={blurHandle}
+             required/>
+      <img className={'image'} alt='eye' src={visiblePassword ? open_eye : close_eye} onClick={showPassword}/>
+      {(isError && isBlur) && <span className={'error'}>Password must contain from 6 to 256 characters</span>}
+    </div>
+  )
```
Comment (NatashaMikhaltsova) [inline src/components/Password.tsx:49]: Should there be Confirm Password field here?

**Review by NatashaMikhaltsova — CHANGES_REQUESTED** [review]

**Review by OxygenO22 — APPROVED** [review]

**Review by Terapool — CHANGES_REQUESTED** [review]

**Review by Terapool — APPROVED** [review]

**Review by DimitarShemet — APPROVED** [review]

File: src/components/Password.tsx (hunk 8 above)
Comment (NatashaMikhaltsova) [reply src/components/Password.tsx]: Discussed on daily meetings

**Review by NatashaMikhaltsova — APPROVED** [review]

**Pull request merged**